scripts/archive_api.py

```diff
@@ -572,10 +848,116 @@ def _is_bytes_like(obj):
 	except:
 		return False
 
-@app.post("/topic/{topic_name}")
+@app.post("/topic/{topic_name}",
+         description="Publish a message directly to the archive. "
+                     "Messages should generally be sent directly via the Kafka brokers, but large "
+                     "messages and 'attachement' data can be uploaded directly via this mechanism, "
+                     "and then referenced in smaller messages sent directly. "
+                     "Authentication is required to publish messages, and authorization is required "
+                     "for writing to the target topic.",
+         response_class=Response,
+         openapi_extra={
+             "requestBody": {
+                 "content": {
+                     "application/bson": {
+                         "schema": {
+                             "type": "object",
+                             "properties": {
+                                 "message": {
+                                     "type": "string",
+                                     "format": "binary",
+                                     "description": "The body of the message."
+                                 },
+                                 "headers": {
+                                     "type": "array",
+                                     "description": "The Kafka headers attached to the message. "
+                                                    "This must be either an array of 2-tuples mapping "
+                                                    "strings to binary blobs, or an equivalent dictionary/object.",
+                                     "items": {
+                                         "type": "array",
+                                         "minItems": 2,
+                                         "maxItems": 2,
```

Comment:
Is this saying the headers must have exactly 2 items in it? If so can you just list the two items in the docs explicitly? It looks like Hermes is using `_id` and `format` in the headers for the files it posts to the archive.